src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -1740,12 +1740,15 @@ public void handle(final Callback[] callbacks) throws IOException, UnsupportedCa
                         continue;
                     }
                     final TrustedAuthoritiesCallback trustedAuthoritiesCallback = (TrustedAuthoritiesCallback) callback;
-                    if (trustedAuthorities == null) {
-                        trustedAuthorities = new ArrayList<>(trustedAuthoritiesCallback.getTrustedAuthorities());
-                    } else {
-                        final List<TrustedAuthority> authorities = new ArrayList<>(trustedAuthoritiesCallback.getTrustedAuthorities());
-                        authorities.removeIf(trustedAuthorities::contains);
-                        trustedAuthorities.addAll(authorities);
+                    List<TrustedAuthority> trustedAuthoritiesHolder = trustedAuthoritiesCallback.getTrustedAuthorities();
+                    if (trustedAuthoritiesHolder != null){
+                        if (trustedAuthorities == null) {
+                            trustedAuthorities = new ArrayList<>(trustedAuthoritiesCallback.getTrustedAuthorities());
```

Comment:
Just a minor comment, this line can use ```trustedAuthoritiesHolder``` instead of ```trustedAuthoritiesCallback.getTrustedAuthorities()```

Reply:
Fixed